src/main/java/io/singularitynet/Client/VereyaModClient.java

```diff
@@ -202,55 +240,44 @@ public void setInputType(InputType input)
     }
 
     private void onKey(long window, int key, int scancode, int action, int modifiers) {
+        // do default thing if any screen is open
+        Screen screen = MinecraftClient.getInstance().currentScreen;
+        if (screen != null) {
+            if (screen instanceof ChatScreen) {
+                return;
+            }
+        }
+
         if ((key == GLFW.GLFW_KEY_F6) && (action == GLFW.GLFW_PRESS))
         {
-            if (inputTypeAbs == InputType.HYBRID)
-            {
-                inputTypeAbs = this.prevInputType;
-                setInputType(this.prevInputType);
-            }
-            else
-            {
-                this.prevInputType = inputTypeAbs;
-                inputTypeAbs = InputType.HYBRID;
-                setInputType(InputType.AI);
-            }
+            setInputType(inputType.hNext());
         }
-        if (inputTypeAbs==InputType.HYBRID) {
-            boolean bKey = (key == GLFW.GLFW_KEY_W) || (key == GLFW.GLFW_KEY_S) || (key == GLFW.GLFW_KEY_A) ||
-                    (key == GLFW.GLFW_KEY_D) || (key == GLFW.GLFW_KEY_SPACE);
-            if (bKey && (action == GLFW.GLFW_PRESS)) {
-                if ((inputType == InputType.AI) & !(MinecraftClient.getInstance().currentScreen instanceof ChatScreen)) {
-                    setInputType(InputType.HUMAN);
-                } else {
-                    return;
-                }
-            } else if (bKey && (action == GLFW.GLFW_RELEASE)) {
-                if (inputType == InputType.HUMAN) {
-                    setInputType(InputType.AI);
-                } else {
-                    return;
-                }
+
+        if (((inputType == InputType.HYBRID_KEYBOARD))) {
+            if ((action == GLFW.GLFW_PRESS)) {
+                MinecraftClient.getInstance().mouse.lockCursor();
```

Comment:
Why mouse is locked in this mode? You said only keyboard is used in this mode.

Reply:
@noskill  removed